go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1402,34 +1390,49 @@ func (tsv *TabletServer) convertAndLogError(ctx context.Context, sql string, bin
 	// 2. FAILED_PRECONDITION errors. These are caused when a failover is in progress.
 	// If so, we don't want to suppress the error. This will allow VTGate to
 	// detect and perform buffering during failovers.
-	if tsv.TerseErrors && len(bindVariables) != 0 && errCode != vtrpcpb.Code_FAILED_PRECONDITION {
-		sqlErr, ok := origErr.(*mysql.SQLError)
-		if ok {
-			sqlState := sqlErr.SQLState()
-			errnum := sqlErr.Number()
-			err = vterrors.Errorf(errCode, "(errno %d) (sqlstate %s) during query: %s", errnum, sqlState, sqlparser.TruncateForLog(sql))
-			err = formatErrorWithCallerID(ctx, err)
+	var message string
+	sqlErr, ok := err.(*mysql.SQLError)
+	if ok {
+		sqlState := sqlErr.SQLState()
+		errnum := sqlErr.Number()
+		if tsv.TerseErrors && len(bindVariables) != 0 && errCode != vtrpcpb.Code_FAILED_PRECONDITION {
+			err = vterrors.Errorf(errCode, "(errno %d) (sqlstate %s)%s during query: %s", errnum, sqlState, callerID, sqlparser.TruncateForLog(sql))
+			message = err.Error()
+		} else {
+			err = vterrors.Errorf(errCode, "%s (errno %d) (sqlstate %s)%s: %s", sqlErr.Message, errnum, sqlState, callerID, queryAsString(sql, bindVariables))
+			message = err.Error()
+		}
+	} else {
+		err = vterrors.Errorf(errCode, "%v%s", err.Error(), callerID)
+		if tsv.TerseErrors && len(bindVariables) != 0 && errCode != vtrpcpb.Code_FAILED_PRECONDITION {
+			message = fmt.Sprintf("%v: %v", err, queryAsString(sql, nil))
```

Comment:
This is a change in behavior, isn't it? We won't log the bind variables in this case, but we were logging them before.

Reply:
Alain is correct... the tests show this change in behavior:

```
 --- FAIL: TestTerseErrorsBindVars (0.46s)
	tabletserver_test.go:2624: error log '(errno 10) (sqlstate HY000) during query: select * from test_table where a = :a', want 'sensitive message (errno 10) (sqlstate HY000) during query: select * from test_table where a = 1: Sql: "select * from test_table where a = :a", BindVars: {a: "type:INT64 value:\"1\" "}'
```

The behavior we want is for the full mysql error message and all bind vars to be logged to the local log, but the stripped error to be returned back over the wire.